logging: Enable LOG_PRINTK if PRINTK is enabled

nordicjm opened the pull request:
This change will enable the logging system to handle printk() calls by default if printk support is enabled, which improves output resiliency when logging, mcumgr and printk calls are used.

Fixes #38947
Fixes #50464

nordicjm committed:
logging: Enable LOG_PRINTK if PRINTK is enabled

This change will enable the logging system to handle printk() calls by
default if printk support is enabled, which improves output resiliency
when logging, mcumgr and printk calls are used. Will disable keep
LOG_PRINTK disabled for tests which are not using the new API as
moving to LOG_PRINTK causes test data to be output to a different UART
in the old test system.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
logging: Increase dummy shell buffer size to 400 bytes for LOG_PRINTK

If LOG_PRINTK is used then the buffer size needs to be larger to
account for the additional header/footer output from the shell.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm commented:
@nordic-krch The tests are failing because of using synchronous logging, if minimal footprint logging is selected then the output appears. I've tried adding `LOG_PANIC()` on the test complete handler and this does not force the output to appear

nordic-krch commented:
there is a `log_api` test in LOG_PRINTK configuration which is working fine so it should be possible to fix those tests. Will try to find the root cause when have some free time.

nordicjm commented:
@nordic-krch Any update on this issue?

nordic-krch committed:
tests: logging: log_core_additional: Align test to LOG_PRINTK changes

Align test to accomodate for a change that LOG_PRITNK is by default
enabled.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: logging: log_syst: Align to LOG_PRINTK as default

Alignging test to a change which enabled LOG_PRINTK as default.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: logging: log_switch_format: Align to LOG_PRINTK as default

Alignging test to a change which enabled LOG_PRINTK as default.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: net: shell: Align to LOG_PRINTK as default

Alignging test to a change which enabled LOG_PRINTK as default.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: openthread: Align to LOG_PRINTK as default

Alignging test to a change which enabled LOG_PRINTK as default.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch commented:
@nordicjm  I tried to fix some tests and in others i simply forced LOG_PRINTK=n to push it forward.

nordic-krch committed:
tests: logging: log_links: Align to LOG_PRINTK as default

Alignging test to a change which enabled LOG_PRINTK as default.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: logging: log_link_order: Align to LOG_PRINTK as default

Alignging test to a change which enabled LOG_PRINTK as default.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
samples: tfm_integration: psa_firmware: Improve logging handling

Remove redundant initialization and add logs flushing before
the reboot.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
doc: logging: Add section about LOG_PRINTK option

Add section which describes coexistence of printk and logging
and the behavior of LOG_PRINTK with information that it is by
default enabled.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>
